clang/lib/Driver/Driver.cpp

```diff
@@ -1126,6 +1127,25 @@ void Driver::CreateOffloadingDeviceToolChains(Compilation &C,
         << SYCLHostCompilerOptions->getSpelling().split('=').first
         << "-fsycl-host-compiler";
 
+  Arg *SYCLUIHArg = C.getInputArgs().getLastArg(
+      options::OPT_fsycl_use_integration_headers,
+      options::OPT_fno_sycl_use_integration_headers);
+  if (SYCLUIHArg) {
+    // -f[no-]sycl-use-integration-headers cannot be used without
+    // -fsycl.  (-fintelfpga implies -fsycl).
+    if (!HasSYCL && !SYCLfpga)
```

Comment:
Is there any reason not to include the test of `SYCLfpga` in the initialization of `HasSYCL` above? As is, it looks like `HasValidSYCLRuntime` will be initialized to `false` if `-fintelfpga` is passed (but `-fsycl` is not).